app/queries/create_quiz_submission_mutator.rb

```diff
@@ -10,21 +11,24 @@ class CreateQuizSubmissionMutator < ApplicationQuery
   validate :ensure_submittability
 
   def create_submission
-    TimelineEvent.transaction do
-      submission = target.timeline_events.create!(
+    submission =
+      TimelineEvent.transaction do
+      target.timeline_events.create!(
         checklist: result[:checklist],
         quiz_score: result[:score],
         passed_at: Time.zone.now
-      )
+      ).tap do |submission|
+        students.map do |student|
+          student.timeline_event_owners.create!(timeline_event: submission, latest: true)
+        end
 
-      students.map do |student|
-        student.timeline_event_owners.create!(timeline_event: submission, latest: true)
+        TimelineEvents::AfterMarkingAsCompleteJob.perform_later(submission)
```

Comment:
This `perform_later` should also be outside the transaction.

Reply:
The same doubts as above ^^^